daemon/container.go

```diff
@@ -251,6 +251,19 @@ func (daemon *Daemon) verifyContainerSettings(hostConfig *containertypes.HostCon
 		}
 	}
 
+	p := hostConfig.RestartPolicy
+
+	switch p.Name {
+	case "always", "unless-stopped":
+		if p.MaximumRetryCount != 0 {
+			return nil, fmt.Errorf("maximum restart count not valid with restart policy of '%s'", p.Name)
+		}
+	case "no", "on-failure", "":
+	// do nothing
```

Comment:
erm, and while we're at it, perhaps we can make the error more friendly if a non-numeric value is provided;

```
docker run --restart no:such-thing -d nginx:alpine
docker: strconv.ParseInt: parsing "such-thing": invalid syntax.
```